cadc-gms/src/main/java/org/opencadc/auth/StandardIdentityManager.java

```diff
@@ -296,86 +333,215 @@ public String toDisplayString(Subject subject) {
-                        // TODO: register an X509 DN with IAM and see if I can get it back here
-
-                        OpenIdPrincipal oip = new OpenIdPrincipal(oidcClient.getIssuerURL(), sub);
-                        HttpPrincipal hp = new HttpPrincipal(username);
-
-                        s.getPrincipals().remove(raw);
-                        s.getPrincipals().add(oip);
-                        s.getPrincipals().add(hp);
-
-                        AuthorizationToken authToken = new AuthorizationToken(challengeType, credentials, oidcDomains, oidcScope);
-                        s.getPublicCredentials().add(authToken);
-                    } catch (NotAuthenticatedException ex) {
-                        JSONObject json = new JSONObject(ex.getMessage());
-                        String error = json.getString("error");
-                        String details = json.getString("error_description");
-                        // details usually includes the invalid access token: truncate
-                        StringBuilder sb = new StringBuilder(error);
-                        sb.append(" reason: ");
-                        int max = Math.min(details.length(), 32);
-                        sb.append(details.subSequence(0, max));
-                        if (max < details.length()) {
-                            sb.append("...");
+        boolean validated = false;
+        for (AuthorizationTokenPrincipal raw : rawTokens) {
+            if (AuthenticationUtil.AUTHORIZATION_HEADER.equalsIgnoreCase(raw.getHeaderKey())) {
+                String[] tval = raw.getHeaderValue().split(" ");
+                if (tval.length == 2) {
+                    if ("Bearer".equalsIgnoreCase(tval[0])) {
+                        validateWithToken(s, raw);
+                        if (!validated) {
+                            validated = true;
+                        } else {
+                            throw new NotAuthenticatedException(raw.getHeaderValue(), NotAuthenticatedException.AuthError.INVALID_REQUEST,
+                                    "Multiple bearer token authorization headers not supported");
                         }
-                        throw new NotAuthenticatedException(challengeType, NotAuthenticatedException.AuthError.INVALID_TOKEN, sb.toString());
-                    } catch (Exception ex) {
-                        throw new NotAuthenticatedException(challengeType, NotAuthenticatedException.AuthError.INVALID_TOKEN, ex.getMessage(), ex);
-                    }
+                    } // else not a bearer token
+                } else {
+                    throw new NotAuthenticatedException(raw.getHeaderValue(), NotAuthenticatedException.AuthError.INVALID_REQUEST,
+                            "BUG: invalid authorization " + raw.getHeaderValue());
                 }
+            } // else other challenge
+        }
+    }
+
+    private void validateWithToken(Subject s, AuthorizationTokenPrincipal raw) {
```

Comment:
This method name could be simply `validateToken` since it does both validateWithPubKey and validateWithUserInfo... the `With` seems slightly misleading as it sounds like "using the token" which is really only the latter method